core/services/ocrcommon/telemetry.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type eaTelemetryResponse struct {
+	DataSource                    string `json:"data_source"`
+	ProviderRequestedProtocol     string `json:"provider_requested_protocol"`
+	ProviderRequestedTimestamp    int64  `json:"provider_requested_timestamp"`
+	ProviderReceivedTimestamp     int64  `json:"provider_received_timestamp"`
+	ProviderDataStreamEstablished int64  `json:"provider_data_stream_established"`
+	ProviderDataReceived          int64  `json:"provider_data_received"`
+	ProviderIndicatedTime         int64  `json:"provider_indicated_time"`
+}
+
+// shouldCollectTelemetry returns whether EA telemetry should be collected
+func shouldCollectTelemetry(jb *job.Job) bool {
+	if jb.Type.String() == pipeline.OffchainReportingJobType && jb.OCROracleSpec != nil {
+		return jb.OCROracleSpec.CaptureEATelemetry
+	}
+
+	if jb.Type.String() == pipeline.OffchainReporting2JobType && jb.OCR2OracleSpec != nil {
+		return jb.OCR2OracleSpec.CaptureEATelemetry
+	}
+
+	return false
+}
+
+// getContract fetches the contract address from the OracleSpec
+func getContract(jb *job.Job) string {
+	switch jb.Type.String() {
+	case pipeline.OffchainReportingJobType:
+		return jb.OCROracleSpec.ContractAddress.String()
+	case pipeline.OffchainReporting2JobType:
+		return jb.OCR2OracleSpec.ContractID
+	default:
+		return ""
+	}
+}
+
+// getChainID fetches the chain id from the OracleSpec
+func getChainID(jb *job.Job) string {
+	switch jb.Type.String() {
+	case pipeline.OffchainReportingJobType:
+		return jb.OCROracleSpec.EVMChainID.String()
+	case pipeline.OffchainReporting2JobType:
+		contract, _ := jb.OCR2OracleSpec.RelayConfig["chainID"].(string)
+		return contract
+	default:
+		return ""
+	}
+}
+
+// parseEATelemetry attempts to parse the bridge telemetry
+func parseEATelemetry(b []byte) (eaTelemetryResponse, error) {
+	type generalResponse struct {
+		Telemetry eaTelemetryResponse `json:"telemetry"`
+	}
+	gr := generalResponse{}
+
+	if err := json.Unmarshal(b, &gr); err != nil {
+		return eaTelemetryResponse{}, err
+	}
+
+	return gr.Telemetry, nil
+}
+
+// getJsonParsedValue checks if the next logical task is of type pipeline.TaskTypeJSONParse and trys to return
+// the response as a *big.Int
+func getJsonParsedValue(trr pipeline.TaskRunResult, trrs *pipeline.TaskRunResults) *big.Int {
+	nextTask := trrs.GetNextTaskOf(trr)
+	if nextTask != nil && nextTask.Task.Type() == pipeline.TaskTypeJSONParse {
+		asDecimal, err := utils.ToDecimal(nextTask.Result.Value)
+		if err != nil {
+			return nil
+		}
+		return asDecimal.BigInt()
+	}
+	return nil
+}
+
+// collectEATelemetry checks if EA telemetry should be collected, gathers the information and sends it for ingestion
+func collectEATelemetry(ds *inMemoryDataSource, trrs *pipeline.TaskRunResults, finalResult *pipeline.FinalResult) {
```

Comment:
Seems like all the work here is blocking (until the last line), any way we can make this fire and forget? or otherwise benchmark impact of this on the Observe call? This can be deferred to a later PR though.